mne_bids/write.py

```diff
@@ -977,37 +987,37 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
     on_invalid_er_task = 'continue'
 
     scans_fname = _gen_bids_basename(
-        subject=subject_id, session=session_id, suffix='scans.tsv',
-        prefix=ses_path, on_invalid_er_task=on_invalid_er_task)
+        subject=subject_id, session=session_id,
+        prefix=ses_path, suffix='scans.tsv',
+        on_invalid_er_task=on_invalid_er_task)
+
     coordsystem_fname = _gen_bids_basename(
-        subject=subject_id, session=session_id, acquisition=acquisition,
-        suffix='coordsystem.json', prefix=data_path,
+        subject=subject_id, session=session_id,
+        acquisition=acquisition,
+        prefix=data_path, suffix='coordsystem.json',
         on_invalid_er_task=on_invalid_er_task)
     electrodes_fname = _gen_bids_basename(
-        subject=subject_id, session=session_id, acquisition=acquisition,
-        suffix='electrodes.tsv', prefix=data_path,
+        subject=subject_id, session=session_id,
+        acquisition=acquisition,
+        prefix=data_path, suffix='electrodes.tsv',
```

Comment:
why don't you do:

```py
electrodes_fname = _gen_bids_basename(bids_path, on_valid_er_task=on_valid_er_task)
```
?

Reply:
Hmm I tried and  then `test_find_matching_sidecar` broke. I think it is because the coordsystem and electrodes files don't need all the other task/run/recording/processing/etc. according to the current specification: https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/02-magnetoencephalography.html#channels-description-_channelstsv

However, that brings up another point: It isn't clear if the template needs to match across different modalities. Right now for coordsystem, subject/session/space/acquisition are generally needed depending on modality. So I added space in.

Reply:
ummm ... not sure why `test_find_matching_sidecar` would fail. Can you perhaps interact with @hoechenberger to make this happen? I think [this](https://github.com/mne-tools/mne-bids/pull/446#discussion_r440344654) is the API we should use with the new object

Reply:
Well I can make it work w/ the following appendage to the unit test. However, the bids specification isn't really clear if say... task and run are needed, or even allowed in coordsystem/electrodes files.